src/service.cc

```diff
@@ -419,8 +410,8 @@ Status Service::genotype_sites(const genotyper_config& cfg, const string& sample
     S(BCFFileSink::Open(cfg, filename, hdr.get(), bcf_out));
 
     // set up the residuals file
-    unique_ptr<Residuals> residuals = nullptr;
-    unique_ptr<ResidualsFile> residualsFile = nullptr;
+    shared_ptr<Residuals> residuals = nullptr;
+    shared_ptr<ResidualsFile> residualsFile = nullptr;
```

Comment:
I'd slightly prefer `residualsFile` remain a `unique_ptr` given that it's an RAII container (the destructor closes the file)...i.e. it should be predictable and explicit when/where it gets destructed